File: factory_pattern_py/ShapeFactory.py
from Circle import Circle
from Rectangle import Rectange
from Square import Square

class ShapeFactory(object):

    #use getShape method to get object of type shape 
    def getShape(self,shapeType):
        """ get the expected object type

        Args:
            shapeType ([type]): [description]

        Returns:
            [type]: [description]
        """
        if(shapeType == None):
            return None
        
        if shapeType.lower() == 'circle':
            return Circle()
        elif shapeType.lower() == 'rectangle':
            return Rectange()
        elif shapeType.lower() == 'square':
            return Square()
        
        return None 

File: factory_pattern_cpp/ShapeFactory.cc
#include "Shape.h"
#include "Rectangle.h"
#include "Square.h"
#include "Circle.h"
#include "ShapeFactory.h"

Shape * ShapeFactory::getShape(ShapeType shapeType){
    Shape * pShape = 0;
    switch (shapeType)
    {
        case ShapeType::Rectangle:
            pShape = new Rectangle();
            break;
        case ShapeType::Square:
            pShape = new Square();
            break;
        case ShapeType::Circle:
            pShape = new Circle();
            break;
        default:
            break; 
    }
    return pShape;
}